apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs

```diff
@@ -618,15 +612,7 @@ private object ConvertToClrTypeImpl(Value wireValue, System.Type targetClrType,
             if (targetClrType == typeof(PgNumeric))
             {
                 bool isPgNumeric = TypeAnnotationCode == TypeAnnotationCode.PgNumeric;
-                bool isFloat64 = TypeCode == TypeCode.Float64;
-                // Check if Float64 is being read as PgNumeric due to user provided type mapping. 
-                bool configuredFloat64AsPgNumeric = isFloat64 && options.Float64ToConfiguredClrType == typeof(PgNumeric); 
-                // For PgNumeric, the value is contained in Value.StringValue.
-                // If the user configures that Float64 should be mapped to PgNumeric CLR type in connection string,
-                // we will need to read Float64 as PgNumeric. The Float64 value is populated in NumberValue of Value.
-                // If the TypeCode is Float64 due to user configured mapping or TypeAnnotationCode is PgNumeric, it is fine.
-                // In all other cases, throw an exception.
-                if (!isPgNumeric && !configuredFloat64AsPgNumeric)
+                if (!isPgNumeric)
```

Comment:
Just change to `if (TypeAnnotationCode != TypeAnnotationCode.PgNumeric)` perhaps?

Reply:
Done.